.github/workflows/tests.yaml

```diff
@@ -74,33 +81,43 @@ jobs:
         if: steps.filter.outputs.has_code == 'true'
         with:
           # Match the min version listed in docs/project/contribution_tools.md
-          python-version: '3.9'
+          # or the oldest version available on the OS.
+          python-version: ${{ matrix.os == 'macos-14' && '3.11' || '3.9' }}
 
       # Install and cache LLVM 16 from Homebrew.
       # TODO: We can potentially remove this and simplify things when the
       # Homebrew version of LLVM updates to 16 here:
       # https://github.com/actions/runner-images/blob/main/images/macos/macos-12-Readme.md
       - name: Cache Homebrew (macOS)
-        if: steps.filter.outputs.has_code == 'true' && matrix.os == 'macos-12'
+        if:
+          steps.filter.outputs.has_code == 'true' && contains(matrix.os,
+          'macos')
         id: cache-homebrew-macos
         uses: actions/cache@v3
         env:
           cache-name: cache-homebrew
         with:
-          # Cover all the critical parts of Homebrew here.
+          # Cover all the critical parts of Homebrew here. Homebrew on Arm macOS
+          # uses its own prefix making this easy to cover, but we need a few
+          # different paths for Intel.
```

Comment:
Just to be sure, this is arm rather than the macos-14 setup?

Reply:
Yep, all of Arm Homebrew uses this layout.

My guess is they made the change to a parallel root because there was a pretty long period of time where the *x86* Homebrew was in use on Arm machines via Rosetta. This "just worked" for a bunch of use cases, and there were a lot of installs like this. When they got ready to have a native Arm Homebrew, they wanted it to be able to be installed in parallel with the x86 one to avoid breakage, so they moved to their own prefix.